### ISSUE TITLE
UnusedPrivateClass: false negative with import directives

### ISSUE THREAD

**Issue by t-kameyama** [detekt/detekt#2816]
## Steps to Reproduce
```kotlin
package com.example

import com.example.C.EFG.EFG1

class C {
    fun test() {
        println(EFG1)
    }

    private enum class E { // false negative
        E1
    }

    private enum class EFG {
        EFG1
    }
}
```

Related: https://github.com/detekt/detekt/pull/2812#discussion_r443204280
